### DISCUSSION TITLE
ESP32S3 BLE Central Crashes After Connection

### DISCUSSION THREAD

**Discussion by gaurav0096** [micropython/micropython#12183] [ESP32]
I am encountering an issue while using the ESP32S3 BLE central example code. After establishes a connection with a peripheral device the Central device got crashed. I have tested this on multiple instances, and the problem persists consistently. 

**Found peripheral: 0 b'|\xdf\xa1\xe4n\xa6' mpy-uart**
*****ERROR*** A stack overflow in task nimble_host has been detected.**
**Backtrace: 0x40376282:0x3fcb5240 0x40386349:0x3fcb5260 0x403891fe:0x3fcb5280 0x40387e7e���**

**Device Specification:-**
ESP-ROM:esp32s3-20210327
Build:Mar 27 2021
rst:0xc (RTC_SW_CPU_RST),boot:0x18 (SPI_FAST_FLASH_BOOT)
Saved PC:0x42060110
SPIWP:0xee
Octal Flash Mode Enabled
For OPI Flash, Use Default Flash Boot Mode
mode:SLOW_RD, clock div:1
load:0x3fce3810,len:0xe68
load:0x403c9700,len:0xa08
load:0x403cc700,len:0x2a44
entry 0x403c987c
MicroPython 01c758e-dirty on 2023-07-30; ESP32S3 module (spiram) with ESP32S3

Central device code
https://github.com/micropython/micropython/blob/master/examples/bluetooth/ble_simple_central.py
Peripheral device code
https://github.com/micropython/micropython/blob/master/examples/bluetooth/ble_simple_peripheral.py

**Comment by cbockermann** [comment]
I can confirm this, happening locally here on two ESP32 devices.
Peripheral code runs fine, detection via scan is fine, as soon as a gap_connect is issued, the exception is thrown.

**Reply by amplatzer** [reply]
try the 1.20 version firmware.
